shared/nm-utils/nm-secret-utils.c

```diff
@@ -132,3 +133,31 @@ nm_secret_buf_to_gbytes_take (NMSecretBuf *secret, gssize actual_len)
 	                                   _secret_buf_free,
 	                                   secret);
 }
+
+/*****************************************************************************/
+
+/**
+ * nm_utils_memeqzero_secret:
+ * @data: the data pointer to check (may be %NULL if @length is zero).
+ * @length: the number of bytes to check.
+ *
+ * Checks that all bytes are zero. This always takes the same amount
+ * of time to prevent timing attacks.
+ *
+ * Returns: whether all bytes are zero.
+ */
+gboolean
+nm_utils_memeqzero_secret (gconstpointer data, gsize length)
+{
+	const guint8 *const key = data;
+	volatile guint8 acc = 0;
+	gsize i;
+
+	/* Based on key_is_zero() by Jason A. Donenfeld <Jason@zx2c4.com>. */
```

Comment:
As mentioned via email, IANAL, but usually in my experience the right way (or at least, the common way) of crediting code that you take from some GPL-2.0 or LGPL-2.1 sources isn't a stray comment like this, but to just take the precise copyright line from the header and put that in the header of this file. On the other hand, it's not as though this is a particularly novel function, so whatever floats your boat...